src/meta/TableSpec.h

```diff
@@ -185,5 +202,52 @@ struct TableSpecEqual {
 
 using TableSpecSet = nebula::common::unordered_set<TableSpecPtr, TableSpecHash, TableSpecEqual>;
 
+constexpr auto HOUR_MINUTES = 60;
+constexpr auto MINUTE_SECONDS = 60;
+constexpr auto DAY_HOURS = 24;
+constexpr auto HOUR_SECONDS = HOUR_MINUTES * MINUTE_SECONDS;
+constexpr auto DAY_SECONDS = HOUR_SECONDS * DAY_HOURS;
+
+const static nebula::common::unordered_map<nebula::meta::PatternMacro, std::string> patternStr{ { nebula::meta::PatternMacro::DATE, "date" }, { nebula::meta::PatternMacro::HOUR, "hour" }, { nebula::meta::PatternMacro::MINUTE, "minute" }, { nebula::meta::PatternMacro::SECOND, "second" }, { nebula::meta::PatternMacro::TIMESTAMP, "timestamp" } };
```

Comment:
to be consistently as `static const`

also could you please wrap each entry into one line? (we want to enforce max line width some day with build rules)

Reply:
We should use UPPER_CASE for static const values, especially namespace level.

Reply:
I think 'date' literal can't be macro, the real macro defined in Nebula already is `{date}`,  it's `%7Bdate%7D` after url encoding.